geniusyield-market-maker/src/GeniusYield/MarketMaker/Prices.hs

```diff
@@ -52,11 +56,84 @@ import           Maestro.Types.V1
 -- >>> let gensAC = "dda5fdb1002f7389b33e036b6afee82a8189becb6cba852e8b79b4fb.0014df1047454e53"
 -- >>> let frenAC = "fc11a9ef431f81b837736be5f53e4da29b9469c983d07f321262ce61.4652454e"
 
-data MaestroPriceException
-  = MaestroPairNotFound
-  | MaestroApiError !Text !MaestroError
-  deriving stock (Show)
-  deriving anyclass (Exception)
+
+-------------------------------------------------------------------------------
+-- Configuration
+-------------------------------------------------------------------------------
+ 
+-- | PriceConfigV1 is deprecated
+data PriceConfigV1 = PriceConfigV1
+  { pcApiKey     ∷ !(Confidential Text),
+    pcResolution ∷ !Resolution,
+    pcNetworkId  ∷ !GYNetworkId,
+    pcDex        ∷ !Dex,
+    pcOverride   ∷ !(Maybe MaestroPairOverride)
+  }
+  deriving stock (Show, Generic)
+  deriving (FromJSON) via CustomJSON '[FieldLabelModifier '[CamelToSnake]] PriceConfigV1
+
+data PriceConfigV2 = PriceConfigV2
+  { pcPriceCommonCfg     ∷ !PriceCommonCfg,
+    pcPricesProviderCfgs ∷ !(NE.NonEmpty PricesProviderCfg)
+  }
+  deriving stock (Show, Generic)
+  deriving (FromJSON) via CustomJSON '[FieldLabelModifier '[CamelToSnake]] PriceConfigV2
+
+data PriceCommonCfg = PriceCommonCfg
+  { pccCommonResolution       ∷ !CommonResolution,
```

Comment:
Likely best to get rid of `CommonResolution` as it imposes the restriction for new price providers.

Reply:
Done.